sound/soc/sof/topology.c

```diff
@@ -2304,6 +2305,19 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 	dai_drv->dobj.private = spcm;
 	list_add(&spcm->list, &sdev->pcm_list);
 
+	ret = sof_parse_tokens(scomp, spcm, stream_tokens,
+			       ARRAY_SIZE(stream_tokens), private->array,
+			       le32_to_cpu(private->size));
+	if (ret) {
+		dev_err(sdev->dev, "error: parse stream tokens failed %d\n",
+			le32_to_cpu(private->size));
+		return ret;
+	}
+
+	dev_vdbg(sdev->dev, "tplg: pcm %s stream tokens: playback d0i3:%d, capture d0i3:%d\n",
+		 spcm->pcm.pcm_name, spcm->stream[0].d0i3_compatible,
+		 spcm->stream[1].d0i3_compatible);
```

Comment:
this is odd, you might log information on a non-existent capture stream if there's playback only?
You should have two logs, one for each direction, and test if they are relevant separately.

Reply:
Makes sense, let me change it.